parser/parser_test.go

```diff
@@ -12,35 +10,50 @@ var repoMap = map[string]string{
 	"lg2": "libgit2/libgit2",
 }
 
-func testExpandRepo(t *testing.T, input string, repo string) {
-	result := Parse(repoMap, input)
-	msg := fmt.Sprintf("input %#v expected repo %#v to be %#v", input, result.Repo, repo)
-	assert.Equal(t, repo, result.Repo, msg)
-}
+func TestParse(t *testing.T) {
+	repoTests := []struct {
+		input string // the input
+		repo  string // the expected repo match or expansion
+	}{
+		{"", ""},                     // no input, no repo match
+		{"df", "zerowidth/dotfiles"}, // match shorthand
+		{" df", ""},                  // no match, leading space
+		{"foo/bar", "foo/bar"},       // fully qualified
+	}
 
-func testExpandRepoIssue(t *testing.T, input string, repo string, issue string) {
-	result := Parse(repoMap, input)
-	msg := fmt.Sprintf("input %#v expected repo %#v to be %#v", input, result.Repo, repo)
-	assert.Equal(t, repo, result.Repo, msg)
-	msg = fmt.Sprintf("input %#v expected issue %#v to be %#v", input, result.Issue, issue)
-	assert.Equal(t, issue, result.Issue, msg)
-}
+	repoIssueTests := []struct {
+		input string // the input
+		repo  string // the expected repo match or expansion
+		issue string // the expected issue match
+	}{
+		{"", "", ""}, // no issue nor repo
```

Comment:
One thing I find useful is to put a `example string` in my test table, which can help suss out which case is failing for many repetitive test cases.

In this case, what you currently have commented would be a good `example`, and I tend to do something like the following for t.Errorf strings:

```
t.Errorf("Parse(repoMap, %s): %s: expected '%s', got '%s'", tc.input, tc.example, tc.repo, result.Repo)
```